packages/langium/src/parser/token-builder.ts

```diff
@@ -49,7 +49,7 @@ export class DefaultTokenBuilder implements TokenBuilder {
 
     protected buildTerminalToken(terminal: TerminalRule): TokenType {
         const regex = terminalRegex(terminal);
-        const pattern = regex.flags.includes('u') ? this.regexPatternFunction(regex) : regex;
+        const pattern = this.requiresCustomPattern(regex) ? this.regexPatternFunction(regex) : regex;
```

Comment:
It does have something to do with Chevrotain, yes. Basically, Chevrotain will apply a view optimizations on token types based on the regex you pass. If you pass a non-trivial regex (usually those with the `/u` flag or lookbehind), those optimizations cannot be performed. Instead of ignoring optimizations, it will instead throw an error. To disable those errors, we wrap the regex inside of an custom pattern function.